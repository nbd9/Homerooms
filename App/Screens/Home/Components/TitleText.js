// @flow
import React from 'react'
import { Text, View, Image } from 'react-native'
import { Images } from '../../../Themes'

import Styles from '../Styles/TitleTextStyles'

type Props = {
  name: string,
  date: string,
  nextSeminar: string
}

const TitleText = ({ name, date, nextSeminar }: Props) => {
  return (
    <View>
      <Image source={Images.school} style={Styles.image} />
      <View style={Styles.imageView}>
        <Text style={Styles.titleText}>Welcome back, {name}</Text>
        <View style={Styles.break} />
        <View>
          <Text style={Styles.summaryText}>Today is {date}</Text>
          <Text style={Styles.summaryText}>The next Support Seminar is {nextSeminar}</Text>
        </View>
      </View>
    </View>
  )
}

export default TitleText
