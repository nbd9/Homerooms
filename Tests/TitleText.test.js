import 'react-native'
import React from 'react'
import TitleText from '../App/Screens/Home/Components/TitleText'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<TitleText name='Test' date='Friday, December 1st' nextSeminar='4 hours' />).toJSON()
  expect(tree).toMatchSnapshot()
})
