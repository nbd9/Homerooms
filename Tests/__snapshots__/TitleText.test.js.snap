// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Image
    source={
      Object {
        "testUri": "../../../App/Images/School.jpg",
      }
    }
    style={
      Object {
        "borderColor": "#3E2723",
        "height": 250,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontFamily": "Lato-Regular",
          "fontSize": 27,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Welcome back, 
      Test
    </Text>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    />
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Today is 
        Friday, December 1st
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        The next Support Seminar is 
        4 hours
      </Text>
    </View>
  </View>
</View>
`;
